### PR TITLE
Update changelog and versions for v0.8.0

### PR DESCRIPTION
**Pull request by benashz**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+ 
+Fix:
+* VaultAuth: set valid status on VaultAuthGlobal deref error: [GH-854](https://github.com/hashicorp/vault-secrets-operator/pull/854)
+* VDS: properly handle the clone cache key variant during client callback execution: [GH-835](https://github.com/hashicorp/vault-secrets-operator/pull/835)
+* Core: delete resource status metrics upon object deletion: [GH-815](https://github.com/hashicorp/vault-secrets-operator/pull/815)
+* VSS: use a constant backoff on some reconciliation errors: [GH-811](https://github.com/hashicorp/vault-secrets-operator/pull/811)
+* VDS: work around Vault DB static creds TTL rollover bug: [GH-730](https://github.com/hashicorp/vault-secrets-operator/pull/730)
+
+Build:
+* CI: bump Vault verisons: [GH-797](https://github.com/hashicorp/vault-secrets-operator/pull/797)
```
Comment (tvoran) [inline CHANGELOG.md:39]: Should we bump the Vault versions again to include Vault 1.17 before release?

**Review by tvoran — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,59 @@
+## 0.8.0 (July 18th, 2024)
+**Important**
+
+* Helm: CRD schema changes are now automatically applied at upgrade time.
+ 
+  *See [updating-crds](https://developer.hashicorp.com/vault/docs/platform/k8s/vso/installation#updating-crds-when-using-helm) for more details.*
+
+* This release contains CRD schema changes which remove the field validation on most VaultAuth spec fields. That means invalid VaultAuth
+  configurations will no longer be handled at resource application time. Please review the VSO logs and K8s
```
[the remark this answers is not in the document]
Comment (benashz) [reply CHANGELOG.md:9]: I left it vague, since the validation can happen on either create or update actions.

File: CHANGELOG.md (hunk 1 above)
Comment (benashz) [reply CHANGELOG.md:39]: Seems reasonable.

Comment (benashz) [reply CHANGELOG.md]: https://github.com/hashicorp/vault-secrets-operator/pull/861

**Pull request merged**